public/plugin.tsx

```diff
@@ -329,7 +333,24 @@ export class AssistantPlugin
         getDisplayName: () => 'Generate visualization',
         getIconType: () => 'visLine' as const,
         execute: async () => {
-          core.application.navigateToApp(TEXT2VIZ_APP_ID);
+          const dataset = data.query.queryString.getQuery().dataset;
+          const url = new URL(core.application.getUrlForApp(TEXT2VIZ_APP_ID, { absolute: true }));
+          if (dataset && dataset.type === 'INDEX_PATTERN') {
```

Comment:
nit: we can replace the hardcoded value with `https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/constants.ts#L53`